Packs/CortexResponseAndRemediation/Playbooks/silent-playbook-Suspicious_Port_Scan.yml

```diff
@@ -0,0 +1,3538 @@
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "159":
+    id: "159"
+    taskid: d02f20bf-d8ab-4cda-b841-8413d251857c
+    type: title
+    task:
+      id: d02f20bf-d8ab-4cda-b841-8413d251857c
+      version: -1
+      name: Block IP
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    nexttasks:
+      '#none#':
+      - "157"
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 939,
+          "y": 3918
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "160":
+    id: "160"
+    taskid: 37ecf6c8-61c3-4a69-8c38-bada08063d6c
+    type: condition
+    task:
+      id: 37ecf6c8-61c3-4a69-8c38-bada08063d6c
+      version: -1
+      name: Is it a high-ratio port scan?
+      description: Checks if the port scan has a highly suspicious connection ratio compared to the baseline.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "1"
+      "yes":
+      - "194"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: isEqualString
+          left:
+            value:
+              simple: alert.name
+            iscontext: true
+          right:
+            value:
+              simple: Highly suspicious port scan
+          ignorecase: true
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 1244,
+          "y": 1352
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "161":
+    id: "161"
+    taskid: 44583cba-46de-4b50-af6e-be223a43cff3
+    type: regular
+    task:
+      id: 44583cba-46de-4b50-af6e-be223a43cff3
+      version: -1
+      name: Search for related alerts to the destination host
+      description: This task searches for related alerts generated by processes within the same causality chain, which may indicate a compromised endpoint.
+      scriptName: SearchAlertsV2
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "141"
+    scriptarguments:
+      extend-context:
+        simple: DstHostRelatedAlerts=
+      fromdate:
+        simple: 3 hours ago
+      ignore-outputs:
+        simple: "true"
+      includeinformational:
+        simple: "true"
+      query:
+        simple: (agentid:${Core.OriginalAlert.event.action_remote_ip} or localip:${Core.OriginalAlert.event.action_remote_ip} or remoteip:${Core.OriginalAlert.event.action_remote_ip}) and -name:"Port Scan" and -name:"Highly suspicious port scan" and -name:"Port scan by suspicious process" and -name:"Suspicious port scan" and -name:"Port Scan With Low Direction Confidence"
```

Comment:
Did you check if it works as expected in cases where one of the agent identifications fails?

Reply:
Of course. This query is based on the agentid, which is always present in these alerts and does not rely on the agent itself. If an agent exists, the agentid corresponds to the agent; otherwise, it contains the IP address or hostname.